tests/e2e/vsphere.go

```diff
@@ -665,6 +665,47 @@ func (vs *vSphere) getHostUUID(ctx context.Context, hostInfo string) string {
 	return ""
 }
 
+//getHostUUID takes input of the HostInfo which has host uuid
+//with the host uuid it maps the corresponding host IP and returns it
+func (vs *vSphere) getAllHostsIP(ctx context.Context) []string {
+	var result []string
+	computeCluster := os.Getenv("CLUSTER_NAME")
+	if computeCluster == "" {
+		if guestCluster {
+			computeCluster = "compute-cluster"
+		} else if supervisorCluster {
+			computeCluster = "wcp-app-platform-sanity-cluster"
+		}
+		framework.Logf("Default cluster is chosen for test")
+	}
+
+	finder := find.NewFinder(vs.Client.Client, false)
+	dc, err := finder.Datacenter(ctx, vs.Config.Global.Datacenters)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	finder.SetDatacenter(dc)
+
+	clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	cluster := clusterComputeResource[0]
+	//TKG setup with NSX has edge-cluster enabled, this check is to skip that cluster
+	if !strings.Contains(cluster.Name(), computeCluster) {
+		cluster = clusterComputeResource[1]
+	}
+
+	config, err := cluster.Configuration(ctx)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	for counter := range config.VsanHostConfig {
+		if config.VsanHostConfig[counter].ClusterInfo.NodeUuid != "" {
```

Comment:
if we do this then the caller has to take care of the empty entries in `hosts`, better to use a different index variable

Reply:
Fixed